tc/c2/convolution_op.h

```diff
@@ -63,8 +63,6 @@ class TcConvolutionOp : public TcOp<T, Context, Engine> {
 
     this->tc_ = tc::makeConvolution2DTc(strideH, strideW);
     this->tcName_ = tc::CONVOLUTION2D_TC_NAME;
-    this->gradTc_ = tc::makeConvolution2DGradTc(strideH, strideW);
-    this->gradTcName_ = tc::CONVOLUTION2D_GRAD_TC_NAME;
```

Comment:
Why is this removal part of adding support for gradient computations?
Did this code do anything useful before?
If not, it should be removed first in a separate commit.

This change also removes the single uses of `CONVOLUTION2D_GRAD_TC_NAME` and `makeConvolution2DGradTc` so they should be removed together with these lines.

Reply:
addressed by introducing a commit that removes all the dead code related to inexistent gradient support. Then @mingzhe09088 's commit introduces proper C2 gradient support.